shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/controler/DatasetApiController.java

```diff
@@ -242,6 +248,25 @@ public ResponseEntity<List<DatasetAndProcessingsDTOInterface>> findDatasetsByIds
 		return new ResponseEntity<>(dtos, HttpStatus.OK);
 	}
 
+	@Override
+	public ResponseEntity<Void> updateDatasetTags(Long datasetId, List<Long> studyTagIds, BindingResult result) throws RestServiceException, EntityNotFoundException, SolrServerException, IOException {
+		Dataset ds = datasetService.findById(datasetId);
+		if (ds == null) {
+			throw new EntityNotFoundException(Dataset.class, datasetId);
+		}
+		for(Long id : studyTagIds){
+			if(!studyTagService.existsById(id)){
```

Comment:
I'm not found of SQL call in loops, maybe you can check if they all exist in one call ?
With something like 
```
Integer countAllByIdIn(Set<Long> ids);

boolean existsAllById(Set<Long> ids) {
    return countAllByIdIn(ids).equals(ids.size());
} 
```

Reply:
But for this,  studyTagIds should be a Set<>

Reply:
Or, as you load them later, just compare the length of the result list with the input ids list's length ?

Reply:
Actually, do we even need to fail here ? Or do we just add th existing one and ignore the others ?